src/uct/ib/rc/verbs/rc_verbs_common.c

```diff
@@ -138,3 +138,98 @@ void uct_rc_verbs_txcnt_init(uct_rc_verbs_txcnt_t *txcnt)
     txcnt->pi = txcnt->ci = 0;
 }
 
+static
+void uct_rc_ep_get_zcopy_handler(uct_rc_iface_send_op_t *op, const void *resp)
+{
+    uct_rc_iface_send_desc_t *desc = ucs_derived_of(op, uct_rc_iface_send_desc_t);
+    char *desc_ptr                 = (char *)(desc + 1);
+    size_t desc_offset             = 0;
+    uct_rc_iface_send_iov_save_desc_t *iov_desc;
+    size_t iov_it, iovcnt, length;
+
+    ucs_assert(NULL != desc->super.buffer);
+    iov_desc = desc->super.buffer;
+    iovcnt   = iov_desc->iovcnt;
+
+    /* copy the payload arrived to the mpool descriptor to the user buffers */
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        length = iov_desc->length[iov_it];
+        memcpy(iov_desc->buffer[iov_it], desc_ptr + desc_offset, length);
+        desc_offset += length;
+    }
+
+    ucs_free(desc->super.buffer);
+    ucs_mpool_put(desc);
+}
+
+ucs_status_t
+uct_rc_iface_verbs_iov_rdma_read_callback(uct_iface_h tl_iface, uint32_t length,
+                                          void **desc_iov_p, size_t *desc_offset_p,
+                                          uint64_t *addr_p, uint32_t *lkey_p)
+{
+    uct_rc_iface_send_desc_t *desc  = *desc_iov_p;
+    char *desc_ptr                  = NULL;
+    uct_rc_iface_send_iov_save_desc_t *iov_desc;
+    uct_rc_iface_t *iface ;
+
+    if (NULL == desc) {
+        iface = ucs_derived_of(tl_iface, uct_rc_iface_t);
+        UCT_RC_IFACE_GET_TX_DESC(iface, &iface->tx.mp, desc);
+        iov_desc = ucs_malloc(sizeof(*iov_desc), "iov_lazy_mem_desc");
```

Comment:
In case of no malloc, we have to place this data into `mpool` descriptor. Placing this data into descriptor makes it different from other `AM\PUT` methods in size. If we change this, we have to implement additional limitation variable in `iface` instead `_bcopy` limit.